### PR TITLE
Fix OR-345: Legacy privacy group does not get added to queryPrivacyGroup

### PR DESCRIPTION
**Pull request by pinges**
Signed-off-by: Stefan Pingel <stefan.pingel@consensys.net>

### CODE REVIEW

File: src/main/java/net/consensys/orion/payload/DistributePayloadManager.java [hunk 1]
```diff
@@ -142,34 +142,27 @@ public void processSendRequest(final SendRequest sendRequest, final Handler<Asyn
...
-          return privacyGroupStorage.put(privacyGroupPayload).thenApply(result -> {
-            queryPrivacyGroupPayload.setPrivacyGroupToAppend(privacyGroupStorage.generateDigest(privacyGroupPayload));
-            return queryPrivacyGroupStorage.update(key, queryPrivacyGroupPayload).thenApply((res) -> {
-              future.complete(privacyGroupPayload);
-              return result;
-            }).exceptionally(e -> {
-              future.fail(new OrionException(OrionErrorCode.ENCLAVE_UNABLE_STORE_PRIVACY_GROUP));
-              return null;
-            });
+      if (privacyGroupStorage.get(privacyGroupStorage.generateDigest(privacyGroupPayload)).get().isPresent()) {
```
Comment (lucassaldanha) [inline src/main/java/net/consensys/orion/payload/DistributePayloadManager.java]: Can we rewrite this to make it easier to read?

**Comment by lucassaldanha** [comment]
It would be good to have a unit test to ensure that we don't inntroduce this bug again.

File: src/acceptance-test/java/net/consensys/orion/acceptance/send/receive/privacyGroup/DualNodesPrivacyGroupsTest.java [hunk 2]
```diff
@@ -292,21 +290,48 @@ void createAndDeleteTwice() {
...
-    deletePrivacyGroupTransaction(firstNode, privacyGroupId, PK_1_B_64);
+    deletePrivacyGroup(firstNode, privacyGroupId, PK_1_B_64);
 
     // try to delete the group again
-    assertThrows(AssertionFailedError.class, () -> deletePrivacyGroupTransaction(firstNode, privacyGroupId, PK_1_B_64));
+    assertThrows(AssertionFailedError.class, () -> deletePrivacyGroup(firstNode, privacyGroupId, PK_1_B_64));
+  }
+
+  @Test
+  void LegacyPrivacyGroupAddedToQueryPrivacyGroupIfNonLegacyGroupHasBeenCreatedBefore() {
```
Comment (lucassaldanha) [inline src/acceptance-test/java/net/consensys/orion/acceptance/send/receive/privacyGroup/DualNodesPrivacyGroupsTest.java]: Nit: start the method name with lowercase :)

Comment (pinges) [reply src/acceptance-test/java/net/consensys/orion/acceptance/send/receive/privacyGroup/DualNodesPrivacyGroupsTest.java]: Ooops
Done!

**Pull request merged**